composer/trainer/trainer.py

```diff
@@ -904,16 +902,19 @@ def __init__(
         log.info('Run name: %s', run_name)
 
         # Create the State
-        self.state = State(rank_zero_seed=rank_zero_seed,
-                           algorithms=algorithms,
-                           model=model,
-                           callbacks=callbacks,
-                           grad_accum=grad_accum,
-                           eval_batch_split=eval_batch_split,
-                           precision=precision,
-                           optimizers=optimizers,
-                           run_name=run_name,
-                           deepspeed_config=deepspeed_config)
+        self.state = State(
+            rank_zero_seed=rank_zero_seed,
+            algorithms=algorithms,
+            model=model,
+            callbacks=callbacks,
+            grad_accum=grad_accum,
+            adaptive_gradient_accumulation=adaptive_gradient_accumulation,
```

Comment:
nit: very long variable name.. maybe `auto_grad_accum` since we use `grad_accum` above? and users set `grad_accum=auto`, not `grad_accum=adaptive`?

regardless, might be confusing for `state` to have both grad accum and auto_grad_accum attributes. trying to think of better naming here.. 🤔